homestar-runtime/src/network/swarm.rs

```diff
@@ -265,30 +277,39 @@ pub(crate) struct ComposedBehaviour {
 
 impl ComposedBehaviour {
     /// Subscribe to [gossipsub] topic.
-    pub(crate) fn gossip_subscribe(&mut self, topic: &str) -> Result<bool, SubscriptionError> {
-        let topic = gossipsub::IdentTopic::new(topic);
-        self.gossipsub.subscribe(&topic)
+    pub(crate) fn gossip_subscribe(&mut self, topic: &str) -> Result<bool> {
+        if let Some(gossipsub) = self.gossipsub.as_mut() {
+            let topic = gossipsub::IdentTopic::new(topic);
+            let subscribed = gossipsub.subscribe(&topic)?;
+
+            Ok(subscribed)
+        } else {
+            Err(anyhow!(NetworkError::PubSubError(PubSubError::NotEnabled)))
```

Comment:
this doesn't have to be anyhow'ed. It could just return NetworkEerror::PubSubError, no? The `?` at the call site should still work, like before.